apps/vaults/components/details/actions/QuickActionsButtons.tsx

```diff
@@ -0,0 +1,165 @@
+	** SWR hook to get the expected out for a given in/out pair with a specific amount. This hook is
+	** called every 10s or when amount/in or out changes. Calls the allowanceFetcher callback.
+	**********************************************************************************************/
+	const	{data: vanillAllowanceFrom, isLoading: isValidatingAllowance, mutate: mutateAllowance} = useSWR(
+		canInteract && !isInputTokenEth && shouldUseVanillaAllowance ?
+		[selectedOptionFrom, spender] as TAllowanceFetcher : null,
+		retrieveAllowance,
+		{revalidateOnFocus: false}
+	);
+
+	useEffect((): void => {
+		if (shouldUseVanillaAllowance && vanillAllowanceFrom) {
+			set_allowanceFrom(vanillAllowanceFrom);
+		}
+	}, [vanillAllowanceFrom, shouldUseVanillaAllowance]);
+
+	const onSuccess = useCallback(async (): Promise<void> => {
+		onChangeAmount(DefaultTNormalizedBN);
+		await refresh();
+	}, [onChangeAmount, refresh]);
+
+	/* 🔵 - Yearn Finance ******************************************************
+	** Trigger an approve web3 action, simply trying to approve `amount` tokens
+	** to be used by the Partner contract or the final vault, in charge of
+	** depositing the tokens.
+	** This approve can not be triggered if the wallet is not active
+	** (not connected) or if the tx is still pending.
+	**************************************************************************/
+	async function	onApproveFrom(): Promise<void> {
+		onApprove(
+			set_txStatusApprove,
+			async (): Promise<void> => {
+				if (currentSolver === Solvers.COWSWAP) {
+					set_allowanceFrom({
+						raw: ethers.constants.MaxUint256,
+						normalized: Infinity
+					});
+				} else {
+					await mutateAllowance();
+				}
+			}
+		);
+	}
+
+	/* 🔵 - Yearn Finance ******************************************************
+	** Trigger a deposit web3 action, simply trying to deposit `amount` tokens to
+	** the selected vault.
+	**************************************************************************/
+	async function	onDepositToVault(): Promise<void> {
+		onExecuteDeposit(
+			set_txStatusDeposit,
+			async (): Promise<void> => {
+				await onSuccess();
+			}
+		);
+	}
+
+	/* 🔵 - Yearn Finance ******************************************************
+	** Trigger a withdraw web3 action using the vault contract to take back
+	** some underlying token from this specific vault.
+	**************************************************************************/
+	async function	onWithdrawShares(): Promise<void> {
+		onExecuteWithdraw(
+			set_txStatusDeposit,
+			async (): Promise<void> => {
+				await onSuccess();
+			}
+		);
+	}
+	
+	/* 🔵 - Yearn Finance ******************************************************
+	** Wrapper to decide if we should use the partner contract or not
+	**************************************************************************/
+	function	onDepositOrWithdraw(): void {
+		if (isDepositing) {
+			onDepositToVault();
+		} else {
+			onWithdrawShares();
+		}
```

Comment:
It would be a bit more readable IMO if
```suggestion
		isDepositing ? onDepositToVault() : onWithdrawShares();
```

Reply:
Looks better without the function here as well, nice 👍

Reply:
Just a follow-up here, https://github.com/yearn/yearn.fi/pull/13#discussion_r1056110587